compass/ocean/tests/baroclinic_gyre/viz.py

```diff
@@ -0,0 +1,140 @@
+import cmocean
+import matplotlib.pyplot as plt
+import numpy as np
+import xarray
+from mpas_tools.cime.constants import constants
+
+from compass.step import Step
+
+
+class Viz(Step):
+    """
+    A step for visualizing output from baroclinic gyre
+
+    Attributes
+    ----------
+    resolution : float
+        The horizontal resolution (km) of the test case
+    """
+    def __init__(self, test_case, resolution):
+        """
+        Create the step
+
+        Parameters
+        ----------
+        test_case : compass.TestCase
+            The test case this step belongs to
+
+        resolution : float
+            The horizontal resolution (km) of the test case
+        """
+        super().__init__(test_case=test_case, name='viz')
+        self.resolution = resolution
+
+    def run(self):
+        """
+        Run this step of the test case
+        """
+
+        out_dir = '.'
+        moc_dir = '../moc'
+        mon_dir = '../forward/output'
+        dsMesh = xarray.open_dataset('../initial_state/initial_state.nc')
+        ds = xarray.open_dataset(f'{moc_dir}/moc.nc')
+        # Insert plots here
+        self._plot_moc(ds, dsMesh, out_dir)
+        self._plot_mean_surface_state(mon_dir, dsMesh, out_dir)
+        self._plot_spinup(mon_dir, dsMesh, out_dir)
+
+    def _plot_moc(self, ds, dsMesh, out_dir):
+        """
+        Plot the time-mean moc state for the test case
+        """
+        avg_len = self.config.getint('mean_state_viz', 'time_averaging_length')
+        moc = ds.moc[-12 * avg_len:, :, :].mean(axis=0).T.values
+        latbins = ds.latBins
+        plt.contourf(
+            latbins, dsMesh.refInterfaces, moc,
+            cmap="RdBu_r", vmin=-12, vmax=12)
+        plt.gca().invert_yaxis()
+        plt.ylabel('Depth (m)')
+        plt.xlabel('Latitude')
+        idx = np.unravel_index(np.argmax(moc), moc.shape)
+        amoc = "max MOC = {:.1e}".format(round(np.max(moc), 1))
+        maxloc = 'at lat = {} and z = {}m'.format(
+            latbins[idx[-1]].values, int(dsMesh.refInterfaces[idx[0]].values))
+        maxval = 'max MOC = {:.1e} at lat={}-{}'.format(
+            round(np.max(moc[:, 175]), 1),
+            latbins[175 - 1].values, latbins[175].values)
```

Comment:
Ideally, break this into several lines where you store these various inputs in variables and then use f-strings rather than the `.format()` methods to put them in place.